lib/ssl/src/dtls_gen_connection.erl

```diff
@@ -621,12 +629,28 @@ handle_alerts([], Result) ->
     Result;
 handle_alerts(_, {stop, _, _} = Stop) ->
     Stop;
-handle_alerts([Alert | Alerts], {next_state, StateName, State}) ->
-     handle_alerts(Alerts, ssl_gen_statem:handle_alert(Alert, StateName, State));
-handle_alerts([Alert | Alerts], {next_state, StateName, State, _Actions}) ->
-     handle_alerts(Alerts, ssl_gen_statem:handle_alert(Alert, StateName, State)).
+handle_alerts(Alerts, {next_state, StateName, State}) ->
+    handle_alerts_or_reset(Alerts, StateName, State);
+handle_alerts(Alerts, {next_state, StateName, State, _Actions}) ->
+    handle_alerts_or_reset(Alerts, StateName, State).
+
+handle_alerts_or_reset([Alert|Alerts], StateName, #state{connection_states = Cs} = State) ->
```

Comment:
can't dtls_connection:alert_or_reset_connection be re-used here as well? it looks like same function apart from instruction under 'undefined' case clause

you're already referencing to dtls_connection:alert_or_reset_connection in this module ...

Reply:
Because it recurses in the undefined case, thought it was the easiest way to handle that.